test/libsolidity/syntaxTests/literalSuffixes/parameters/overloading_integer_vs_integer.sol

```diff
@@ -0,0 +1,21 @@
+function uSuffix(uint8) pure suffix returns (uint) {}
+function uSuffix(uint16) pure suffix returns (int) {}
+
+function iSuffix(int8) pure suffix returns (uint) {}
+function iSuffix(int16) pure suffix returns (int) {}
+
+function iuSuffix(uint8) pure suffix returns (int) {}
+function iuSuffix(int8) pure suffix returns (uint) {}
+
+contract C {
+    int a = 1024 uSuffix;
+    int b = 1024 iSuffix;
+
+    int c = -1024 uSuffix;
+    int d = -1024 iSuffix;
```

Comment:
here we will always receive a positive 1024 inside the `iSuffix` function, right? Maybe there is never a use-case that we will be able to receive an actual negative int?

[``-123 suffix`` is equivalent to ``-(123 suffix)``, so ``suffix`` does not receive ``-123`` as input](https://github.com/ethereum/solidity/pull/14121/files#diff-85812ff9484018acfc04b0dc727b772bded418bb644ba7e1286b59e20d481481R148)

Reply:
ok, maybe if the suffix function is called directly e.g. `iSuffix(-1024)` but normally it will not really happen, right?

Reply:
if we disallow signed int parameters. a user may checkup the corresponding section in the docs.. also we could generate a nice error, explaining this.

Reply:
Yes, we'll always receive 1024, but this is not something we should be assuming in the PR. We should also test it.

Reply:
why not forbid signed integer for suffix functions in general?

Reply:
See https://github.com/ethereum/solidity/pull/12656#discussion_r1169010129.

> is there any case where someone could use the suffix functions really as a suffix and receive a negative number there?

Yes, it's perfectly valid to return a signed value that can be negated. Or a value of an UDVT type that supports `-` operator.